apps/s_client.c

```diff
@@ -1980,6 +2025,21 @@ int s_client_main(int argc, char **argv)
     if (enable_pha)
         SSL_set_post_handshake_auth(con, 1);
 
+#ifndef OPENSSL_NO_RPK
+    if (enable_client_rpk)
+        SSL_set_options(con, SSL_OP_RPK_CLIENT);
```

Comment:
This is the enable/disable configuration. Multiple certificate types are already supported. 

EDIT: As of right now, there's only two possible values that can be part of the CertificateType Extension: RPK and X509. So, for future-proofing, it might be better to have a list than having a single bit. However, I would want it kept relatively easy-to-use, because the OpenSSL APIs are becoming more-and-more complex.

So, perhaps what's not clear is that RPK _does not disable certificates_. It primarily enables the extension.

Endpoints are able to still  configure a certificate. If RPK is negotiated, then the server will send the RPK from the certificate rather than the the certificate itself. If the endpoint did not configure a certificate, but only configured a private key, then that would enable RPK only in the extension.

Admittedly, it's implicit; but it avoids failures in connecting due to bad configuration.

Reply:
> This is the enable/disable configuration. Multiple certificate types are already supported.
> 
> EDIT: As of right now, there's only two possible values that can be part of the CertificateType Extension: RPK and X509. So, for future-proofing, it might be better to have a list than having a single bit. However, I would want it kept relatively easy-to-use, because the OpenSSL APIs are becoming more-and-more complex.

Yes, I figured that out later in a closer review of the code, however, I still object to adding more SSL_OP bits here. There are only so many, and they're already misused for controls that should live elsewhere.  We no longer encourage configuring enabled protocols by setting bit fields, we instead have min/max ranges.  Here there really does need to be a set of supported types.  If you want to provide an `SSL_enable_rpk()` API that does all the work, that's fine, but underneath all that should be a list of supported cert types and add/set interfaces.